tests/functional_tests/ffront_tests/test_execution.py

```diff
@@ -720,6 +719,127 @@ def conditional_program(
+
+    @field_operator
+    def conditional_tuple_2_field_op(
+        mask: Field[[IDim], bool], a: Field[[IDim], float64], b: Field[[IDim], float64]
+    ) -> tuple[tuple[Field[[IDim], float64], Field[[IDim], float64]], Field[[IDim], float64]]:
+        return where(mask, ((3.0, 6.0), a), ((2.0, 7.0), b))
+
+
+def test_conditional_nested_tuple():
+    size = 10
+    mask = np_as_located_field(IDim)(np.zeros((size,), dtype=bool))
+    mask.array()[0 : (size // 2)] = True
+    a = np_as_located_field(IDim)(np.ones((size,)))
+    b = np_as_located_field(IDim)(np.ones((size,)))
+    c = np_as_located_field(IDim)(np.zeros((size,)))
+    d = np_as_located_field(IDim)(np.zeros((size,)))
+
+    @field_operator
+    def conditional_tuple_3_field_op(
+        mask: Field[[IDim], bool], a: Field[[IDim], float64], b: Field[[IDim], float64]
+    ) -> tuple[
+        tuple[Field[[IDim], float64], Field[[IDim], float64]],
+        tuple[Field[[IDim], float64], Field[[IDim], float64]],
+    ]:
+        return where(mask, ((a, b), (b, a)), ((5.0, 7.0), (7.0, 5.0)))
+
+    @program
+    def conditional_tuple_3_p(
+        mask: Field[[IDim], bool],
+        a: Field[[IDim], float64],
+        b: Field[[IDim], float64],
+        c: Field[[IDim], float64],
+        d: Field[[IDim], float64],
+    ):
+        conditional_tuple_3_field_op(mask, a, b, out=((c, d), (d, c)))
+
+    conditional_tuple_3_p(mask, a, b, c, d, offset_provider={})
+
+    assert np.allclose(
+        np.where(
+            mask,
+            ((a, b), (b, a)),
+            ((np.full(size, 5.0), np.full(size, 7.0)), (np.full(size, 7.0), np.full(size, 5.0))),
+        ),
+        ((c, d), (d, c)),
+    )
+
+
+def test_conditional_nested_tuple_mixed_dims():
+    size = 10
+    mask = np_as_located_field(IDim)(np.zeros((size,), dtype=bool))
+    mask.array()[0 : (size // 2)] = True
+    a = np_as_located_field(IDim, JDim)(np.ones((size, size)))
+    b = np_as_located_field(IDim)(np.ones((size,)))
+    c = np_as_located_field(IDim, JDim)(np.zeros((size, size)))
+    d = np_as_located_field(IDim)(np.zeros((size,)))
+    e = np_as_located_field(IDim)(np.zeros((size,)))
+    f = np_as_located_field(IDim)(np.zeros((size,)))
+
+    @field_operator
+    def conditional_tuple_4_field_op(
+        mask: Field[[IDim], bool], a: Field[[IDim, JDim], float64], b: Field[[IDim], float64]
+    ) -> tuple[
+        tuple[Field[[IDim, JDim], float64], Field[[IDim], float64]],
+        tuple[Field[[IDim], float64], Field[[IDim], float64]],
+    ]:
+        return where(mask, ((a, 9.0), (b, 6.0)), ((8.0, b), (5.0, 9.0)))
+
+    @program
+    def conditional_tuple_4_p(
+        mask: Field[[IDim], bool],
+        a: Field[[IDim, JDim], float64],
+        b: Field[[IDim], float64],
+        c: Field[[IDim, JDim], float64],
+        d: Field[[IDim], float64],
+        e: Field[[IDim], float64],
+        f: Field[[IDim], float64],
+    ):
+        conditional_tuple_4_field_op(mask, a, b, out=((c, d), (e, f)))
```

Comment:
The reason you had to touch embedded is not because of how where behave, but because of your test setup here. All out fields should have the same dimensions.

One option would be to take the result of the where and add all the tuple elements into a single out field if you want to keep that test. But I don't think this tests is actually testing something extra that is not already tested (maybe I am missing the point).